features/mbedtls/platform/inc/mbedtls_config_reader.h

```diff
@@ -0,0 +1,500 @@
+    #define MBEDTLS_ECP_DP_CURVE25519_ENABLED
+    #define MBEDTLS_ECP_NIST_OPTIM
+#endif
+
+#ifdef MBEDTLS_CONFIG_GCM
+    #define MBEDTLS_GCM_C
+#endif
+
+#ifdef MBEDTLS_CONFIG_SHA256
+    #define MBEDTLS_SHA256_C
+    #define MBEDTLS_MD_C
+#endif
+
+#ifdef MBEDTLS_CONFIG_SHA512
+    #define MBEDTLS_SHA512_C
+    #define MBEDTLS_MD_C
+#endif
+
+#ifdef MBEDTLS_CONFIG_CBC
+    #define MBEDTLS_CIPHER_MODE_CBC
+    #define MBEDTLS_CIPHER_PADDING_PKCS7
+#endif
+#ifdef MBEDTLS_CONFIG_CCM
+    #define MBEDTLS_CCM_C
+#endif
+
+#ifdef MBEDTLS_CONFIG_PSK
+    #define MBEDTLS_KEY_EXCHANGE_PSK_ENABLED
+#endif
+
+#ifdef MBEDTLS_CONFIG_ECDH
+    #define MBEDTLS_X509_USE_C
+    #define MBEDTLS_PEM_PARSE_C
+    #define MBEDTLS_BASE64_C
+    #define MBEDTLS_ASN1_PARSE_C
+    #define MBEDTLS_PK_PARSE_C
+    #define MBEDTLS_PK_C
+    #define MBEDTLS_ECDH_C
+    #define MBEDTLS_ECP_C
+    #define MBEDTLS_BIGNUM_C
+    #define MBEDTLS_ECP_DP_SECP256R1_ENABLED
+    #define MBEDTLS_ECP_DP_SECP384R1_ENABLED
+    #define MBEDTLS_ECP_DP_CURVE25519_ENABLED
+    #define MBEDTLS_ECP_NIST_OPTIM
+#endif
+
+#ifdef MBEDTLS_CONFIG_ECDH_ECDSA_KEY_EXCHANGE
+    #define MBEDTLS_KEY_EXCHANGE_ECDH_ECDSA_ENABLED
+    #define MBEDTLS_X509_CRT_PARSE_C
+    #define MBEDTLS_SSL_SERVER_NAME_INDICATION
+    #define MBEDTLS_X509_USE_C
+    #define MBEDTLS_PEM_PARSE_C
+    #define MBEDTLS_BASE64_C
+#endif
+
+#ifdef MBEDTLS_CONFIG_RSA
+    #define MBEDTLS_RSA_C
+    #define MBEDTLS_BIGNUM_C
+    #define MBEDTLS_OID_C
+#endif
+
+#ifdef MBEDTLS_CONFIG_ECDHE_RSA_KEY_EXCHANGE
+    #define MBEDTLS_KEY_EXCHANGE_ECDHE_RSA_ENABLED
+    #define MBEDTLS_ECDH_C
+    #define MBEDTLS_RSA_C
+    #define MBEDTLS_PKCS1_V15
+    #define MBEDTLS_PKCS1_V21
+    #define MBEDTLS_X509_CRT_PARSE_C
+    #define MBEDTLS_SSL_SERVER_NAME_INDICATION
+    #define MBEDTLS_X509_USE_C
+    #define MBEDTLS_PEM_PARSE_C
+    #define MBEDTLS_BASE64_C
+#endif
+
+#ifdef MBEDTLS_CONFIG_RSA_KEY_EXCHANGE
+    #define MBEDTLS_KEY_EXCHANGE_RSA_ENABLED
+    #define MBEDTLS_RSA_C
+    
+    #define MBEDTLS_PKCS1_V15
+    #define MBEDTLS_X509_CRT_PARSE_C
+    #define MBEDTLS_SSL_SERVER_NAME_INDICATION
+    #define MBEDTLS_X509_USE_C
+    #define MBEDTLS_PEM_PARSE_C
+    #define MBEDTLS_BASE64_C
+    #define MBEDTLS_BIGNUM_C
+    #define MBEDTLS_OID_C
+    #define MBEDTLS_ASN1_PARSE_C
+    #define MBEDTLS_PK_PARSE_C
+    #define MBEDTLS_PK_C
+#endif
+
+#ifdef MBEDTLS_CONFIG_ECDH_RSA_KEY_EXCHANGE
+    #define MBEDTLS_KEY_EXCHANGE_ECDH_RSA_ENABLED
+    #define MBEDTLS_PKCS1_V15
+    #define MBEDTLS_ECDH_C
+    #define MBEDTLS_X509_CRT_PARSE_C
+    #define MBEDTLS_SSL_SERVER_NAME_INDICATION
+#endif
+
+#define MBED_TLS_HAVE_ASM
+#define MBED_TLS_HAVE_TIME
+#define MBEDTLS_HAVE_TIME
+#define MBEDTLS_REMOVE_ARC4_CIPHERSUITES
+#define MBEDTLS_CIPHER_C
+#define MBEDTLS_ERROR_STRERROR_DUMMY
+#define MBEDTLS_X509_CHECK_KEY_USAGE
+#define MBEDTLS_X509_CHECK_EXTENDED_KEY_USAGE
+#define MBEDTLS_VERSION_C
+#define MBEDTLS_SELF_TEST
```

Comment:
What is the purpose of this "default" options? Are they meant for prototyping? production? something else? For example, what is the point of enabling `MBEDTLS_SELF_TEST` in a production build? Or the `MBEDTLS_VERSION_C` could probably be removed depending on what you want. How about `MBEDTLS_DEBUG_C` and `MBEDTLS_ERROR_C`?

Reply:
They are meant for prototyping, and they are included to allow the default configuration provided by the new system to be as close as possible to the default configuration as before.
For production, we expect people to customize the configuration using a header file to achieve optimum code size/performance/ whatever metric they choose.